pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller_test.go

```diff
@@ -0,0 +1,968 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loadbalancerset
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/openyurtio/openyurt/cmd/yurt-manager/names"
+	"github.com/openyurtio/openyurt/pkg/apis"
+	"github.com/openyurtio/openyurt/pkg/apis/apps/v1beta1"
+	"github.com/openyurtio/openyurt/pkg/apis/network"
+	"github.com/openyurtio/openyurt/pkg/apis/network/v1alpha1"
+)
+
+const (
+	mockServiceName   = "test"
+	mockNodePoolLabel = "app=deploy"
+	mockServiceUid    = "c0af506a-7096-4ef9-b39a-eac2feb5c07g"
+	mockNodePoolUid   = "f47dd9db-d3bc-40f3-8d03-7409930b6289"
+)
+
+var (
+	elbClass = "elb"
+)
+
+func TestReconcilePoolService_Reconcile(t *testing.T) {
+	scheme := initScheme(t)
+
+	t.Run("test get service not found", func(t *testing.T) {
```

Comment:
please add test case with `nodepool-labelselector` is empty.

Reply:
done